src/mlpack/core/data/save_impl.hpp

```diff
@@ -283,110 +283,52 @@ bool Save(const std::string& filename,
-  {
-    std::ostringstream oss;
-    oss << "File type " << Extension(filename) << " not supported.\n";
-    oss << "Currently it supports ";
-    for (auto extension : saveFileTypes)
-      oss << ", " << extension;
-    oss << std::endl;
-    throw std::runtime_error(oss.str());
-    return false;
-  }
-  if (matrix.n_cols > 1)
-  {
-    std::cout << "Input Matrix contains more than 1 image." << std::endl;
-    std::cout << "Only the firstimage will be saved!" << std::endl;
-  }
-  stbi_flip_vertically_on_write(transpose);
-
-  bool status = false;
-  try
-  {
-    unsigned char* image = matrix.memptr();
-
-    if ("png" == Extension(filename))
-    {
-      status = stbi_write_png(filename.c_str(), tempWidth, tempHeight,
-          tempChannels, image, tempWidth * tempChannels);
-    }
-    else if ("bmp" == Extension(filename))
-    {
-      status = stbi_write_bmp(filename.c_str(), tempWidth, tempHeight,
-          tempChannels, image);
-    }
-    else if ("tga" == Extension(filename))
-    {
-      status = stbi_write_tga(filename.c_str(), tempWidth, tempHeight,
-          tempChannels, image);
-    }
-    else if ("hdr" == Extension(filename))
-    {
-      status = stbi_write_hdr(filename.c_str(), tempWidth, tempHeight,
-          tempChannels, reinterpret_cast<float*>(image));
-    }
-    else if ("jpg" == Extension(filename))
-    {
-      status = stbi_write_jpg(filename.c_str(), tempWidth, tempHeight,
-          tempChannels, image, tempQuality);
-    }
-  }
-  catch (std::exception& e)
-  {
-    Timer::Stop("saving_image");
-    if (fatal)
-      Log::Fatal << e.what() << std::endl;
-    Log::Warn << e.what() << std::endl;
-    return false;
-  }
-  Timer::Stop("saving_image");
-  return status;
+  // Call out to .cpp implementation.
+  return SaveImage(filename, tmpMatrix, info, fatal);
 }
 
 // Image saving API for multiple files.
 template<typename eT>
 bool Save(const std::vector<std::string>& files,
           arma::Mat<eT>& matrix,
           ImageInfo& info,
-          const bool fatal,
-          const bool transpose)
+          const bool fatal)
 {
   if (files.size() == 0)
   {
-    std::ostringstream oss;
-    oss << "Files vector is empty." << std::endl;
+    if (fatal)
+    {
+      Log::Fatal << "Save(): vector of image files is empty; nothing to save."
+          << std::endl;
+    }
+    else
+    {
+      Log::Warn << "Save(): vector of image files is empty; nothing to save."
+          << std::endl;
+    }
 
-    throw std::runtime_error(oss.str());
     return false;
   }
-  // We transpose by default. So, un-transpose if necessary.
-  if (!transpose)
-    matrix = arma::trans(matrix);
 
   arma::Mat<unsigned char> img;
-  bool status = Save(files[0], img, info, fatal, transpose);
+  bool status = Save(files[0], img, info, fatal);
 
   // Decide matrix dimension using the image height and width.
   matrix.set_size(info.Width() * info.Height() * info.Channels(), files.size());
```

Comment:
While saving matrix or images, we should resize them , It is probably an error, Since the code might have been copy pasted from Load. 

Matrix is having the same size and then we just have to save them , So Line 334 and 335 is not need, Let me know if I have overlooked something

Reply:
ok, I have handled everything, Made all apt changes, Pulled your branch and rebuild the PR, PTAL when free :)